src/main/java/shortestpath/Transport.java

```diff
@@ -95,48 +95,46 @@ public class Transport {
 
     /** If this is an item transport, this tracks if it is consumable (as opposed to having infinite uses) */
     @Getter
-    private final boolean isConsumable;
+    private boolean isConsumable = false;
 
     /** If this is an item transport, this is the maximum wilderness level that it can be used in */
     @Getter
-    private final int maxWildernessLevel;
+    private int maxWildernessLevel = -1;
 
     /** Any varbits to check for the transport to be valid. All must pass for a transport to be valid */
     @Getter
-    private final List<TransportVarbit> varbits;
+    private final List<TransportVarbit> varbits = new ArrayList<>();
 
     Transport(final WorldPoint origin, final WorldPoint destination) {
         this.origin = origin;
         this.destination = destination;
-        this.isConsumable = false;
-        this.maxWildernessLevel = -1;
-        this.varbits = new ArrayList<>();
     }
 
-    Transport(final String line, TransportType transportType) {
+    Transport(final Map<String, String> fieldMap, TransportType transportType) {
```

Comment:
This function was the main focus. Changing the input form the string of each row of the tsv files then splitting on tabs to map of each row from the tsv.

The old way required every tsv to have the same columns in the same order. This meant the 4th column (`parts[3]` in the code) had to the skill requirements. The map way means this column can be anywhere in the tsv or not present at all.